src/pudl_archiver/archivers/msha.py

```diff
@@ -0,0 +1,42 @@
+"""Download MSHA data."""
+
+import logging
+import re
+import typing
+from pathlib import Path
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+BASE_URL = "https://arlweb.msha.gov/OpenGovernmentData/"
+BROWSER_EXT = "OGIMSHA.asp"
+
+
+class MshaArchiver(AbstractDatasetArchiver):
+    """MSHA archiver."""
+
+    name = "mshamines"
+
+    async def get_resources(self) -> ArchiveAwaitable:
```

Comment:
I think to get the metadata `.txt` definition files the regex would be something like `definition_file_pattern = re.compile(r"([a-zA-Z]+).txt")`. The download link is a little more confusing because sometimes the hyperlink is `DataSets/{Dataset_Name}_Definition_File}.txt` and sometimes it's `/OpenGovernmentData/DataSets/{Dataset_Name}_Definition_File.txt`. So some logic might need to be added here.

and then the download url should be something like `url=BASE_URL + link.split("/")[1:]`